cmd/mdatagen/templates/match.go.tmpl

```diff
@@ -0,0 +1,234 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package metadata
+
+import (
+	"errors"
+	"io"
+	"sort"
+	"strings"
+)
+
+var ErrNotMultimatch = errors.New("this index doesn't represent a valid multimatch pattern")
```

Comment:
I don't see a reason to export any API in this file. We should only export config field types and `Unmarshal` methods